test/e2e/e2e.go

```diff
@@ -130,6 +131,20 @@ func RunE2ETests(t *testing.T) {
 	ginkgo.RunSpecsWithDefaultAndCustomReporters(t, "Kubernetes e2e suite", r)
 }
 
+var suiteCfg types.SuiteConfig
+var reportCfg types.ReporterConfig
+
+var _ = ginkgo.BeforeEach(func() {
+	suiteCfg = types.SuiteConfig{}
+	reportCfg = types.ReporterConfig{}
```

Comment:
This looks very odd. Why does instantiating these types and assigning them to global variables that aren't used anywhere have an effect? And why in a BeforeEach, i.e. after ginkgo has already started executing tests?

The migration guide instead points towards `GinkgoConfiguration()`:

https://onsi.github.io/ginkgo/MIGRATING_TO_V2#changed-availability-of-ginkgos-configuration

Reply:
> assigning them to global variables that aren't used anywhere have an effect 

I remember those variables does used somewhere, for example, verbose configuration, it has been an while I will come back to check it again.

> GinkgoConfiguration 

I will take a look at this, thanks!

Reply:
I'm looking at this PR commit-by-commit, and this commit here adds the variables and writes to them, but then they are not read.

Perhaps some future commit will read them? I haven't gone through all of them. In that case that future commit probably should be merged with this one here, because on its own it doesn't make sense (or needs more explanations).

Reply:
> but then they are not read.

they are not always explicitly read here, but they will be read by Ginkgo instead, e.g. all those bool configuration.

Reply:
Read how? Ginkgo doesn't know about `suiteCfg` and `reportCfg`.

Reply:
@chendave: this still needs to be investigated.

Reply:
I think that pohly is right and we should just follow the migration guide, in `test/e2e/framework/test_context.go` do

```
	suiteConfig, reporterConfig := GinkgoConfiguration()
	suiteConfig.EmitSpecProgress = true
	suiteConfig.RandomizeAllSpecs = true
	reporterConfig.Verbose = true
```

and then just cascade it down to the RunSpecs statements

```go
	// pass it in to RunSpecs
	RunSpecs(t, "My Suite", suiteConfig, reporterConfig)

```

Reply:
We can go with this, but if we go with this, we need to implement the custom report `NewProgressReporter`, and call `RunSpecs` instead of `RunSpecsWithDefaultAndCustomReporters`.

I am not sure if that is possible because that report needs the real-time reporting capabilities while I don't many confidence to achieve that by `ReportBeforeEach` and `ReportAfterEach`

Reply:
Done, pls see: https://github.com/kubernetes/kubernetes/pull/109111/commits/b779bcf5471acd8e65160e373626293d89083f94